torch/csrc/jit/passes/remove_exceptions.cpp

```diff
@@ -0,0 +1,49 @@
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/remove_exceptions.h>
+
+#include <torch/csrc/jit/jit_log.h>
+
+namespace torch {
+namespace jit {
+
+bool certainlyThrows(Block* block) {
+  for (Node* n : block->nodes()) {
+    if (n->kind() == prim::RaiseException) {
```

Comment:
I'm not quite grokking the big picture here, but I would guess it will be used in cases where a block would contain *only* `raise Exception` and *no* statements capable of altering the global state?

Reply:
The assumption here is that the model doesn't throw exceptions and thus `prim::RaiseException` can be used as an indicator of a never executed block. I will add a comment covering that, and hopefully it would address your other concern too by making the purpose of the pass more clear.